src/main/java/org/brapi/test/BrAPITestServer/controller/core/TrialsApiController.java

```diff
@@ -110,6 +99,23 @@ public ResponseEntity<TrialSingleResponse> trialsTrialDbIdGet(@PathVariable("tri
 		Trial data = trialService.getTrial(trialDbId);
 		return responseOK(new TrialSingleResponse(), data);
 	}
+    @CrossOrigin
+	@Override
+	public ResponseEntity<TrialSingleResponse> trialsTrialDbIdDelete(
+			@PathVariable("trialDbId") String trialDbId,
+			@Valid @RequestParam(value = "hardDelete", defaultValue = "false", required = false) boolean hardDelete,
+			@RequestHeader(value = "Authorization", required = false) String authorization) throws BrAPIServerException {
+		log.debug("Request: " + request.getRequestURI());
+		validateSecurityContext(request, "ROLE_USER");
+		validateAcceptHeader(request);
+		 if (hardDelete) {
```

Comment:
The delete cascade works for the use cases of the story and reduces the client code required to carry out a delete.

Any implementation of dynamically determining whether to cascade the delete will increase the scope of work which is difficult to justify without a use case.
If you delete, for example, a trial and not the study then the study record would have a foreign key that points to a non-existing record. There doesn't appear to be non-null constraint on foreign keys, so the code could set any orphan keys to null, but it's hard to see a case where that is desirable.

I would say in the future, if it is necessary to have this feature, the easiest way to implement it would be to create a second soft-delete field that doesn't cascade when the state changes, so any orphan foreign keys at least point to existing records, reducing the potential for bugs.

Reply:
I do agree it would increase the scope of work. I think if you returned an error when attempting to delete an object with child objects it would be possible to avoid the foreign key issue but it would require lookups. I'm fine leaving this for possible future work. If we don't end up making changes, when the brapi docs are updated I think we should make the cascade behavior clear.